### ISSUE TITLE
Effects sorting error -- comparison method violates its general contract

### ISSUE THREAD

**Issue by chapmanb** [pcingola/SnpEff#137]
Pablo;
We're running into an issue with snpEff 4.2 when adding effects to larger structure variants. The effects sort step here:

https://github.com/pcingola/SnpEff/blob/master/src/main/java/org/snpeff/outputFormatter/VcfOutputFormatter.java#L78

Errors out with `Comparison method violates its general contract!` similar to #8 where it was diagnosed as being due to inconsistent comparisons.

Here is a small reproducible test case with a single variant:

https://s3.amazonaws.com/chapmanb/testcases/snpeff_debug_sv_effectsort.tar.gz

and the full traceback:
```
Error: Error while processing VCF entry (line 3417) :
        chr14   106062103       8361    N       <DEL>   .       ReadCountSupport        SVTYPE=DEL;SVLEN=-475711;END=106537814;STRANDS=+-:4;IMPRECISE;CIPOS=-10,468;CIEND=-484,9;CIPOS95=-1,141;CIEND95=-154,1;SU=4;PE=4;SR=0;AC=0;AN=0    GT:SU:PE:SR:GQ:SQ:GL:DP:RO:AO:QR:QA:RS:AS:RP:AP:AB       0/0:4:4:0:102.5:0.0:-9.0,-20.0,-93.0:122:109:13:109:13:0:0:109:13:0.11
java.lang.IllegalArgumentException: Comparison method violates its general contract!
java.lang.IllegalArgumentException: Comparison method violates its general contract!
        at java.util.ComparableTimSort.mergeHi(ComparableTimSort.java:835)
        at java.util.ComparableTimSort.mergeAt(ComparableTimSort.java:453)
        at java.util.ComparableTimSort.mergeForceCollapse(ComparableTimSort.java:392)
        at java.util.ComparableTimSort.sort(ComparableTimSort.java:191)
        at java.util.ComparableTimSort.sort(ComparableTimSort.java:146)
        at java.util.Arrays.sort(Arrays.java:472)
        at java.util.Collections.sort(Collections.java:155)
        at ca.mcgill.mcb.pcingola.outputFormatter.VcfOutputFormatter.addInfo(VcfOutputFormatter.java:78)
        at ca.mcgill.mcb.pcingola.outputFormatter.VcfOutputFormatter.toString(VcfOutputFormatter.java:285)
        at ca.mcgill.mcb.pcingola.outputFormatter.OutputFormatter.endSection(OutputFormatter.java:112)
        at ca.mcgill.mcb.pcingola.outputFormatter.VcfOutputFormatter.endSection(VcfOutputFormatter.java:229)
        at ca.mcgill.mcb.pcingola.outputFormatter.OutputFormatter.printSection(OutputFormatter.java:145)
        at ca.mcgill.mcb.pcingola.snpEffect.commandLine.SnpEffCmdEff.annotate(SnpEffCmdEff.java:278)
        at ca.mcgill.mcb.pcingola.snpEffect.commandLine.SnpEffCmdEff.annotateVcf(SnpEffCmdEff.java:446)
        at ca.mcgill.mcb.pcingola.snpEffect.commandLine.SnpEffCmdEff.annotate(SnpEffCmdEff.java:138)
        at ca.mcgill.mcb.pcingola.snpEffect.commandLine.SnpEffCmdEff.run(SnpEffCmdEff.java:984)
        at ca.mcgill.mcb.pcingola.snpEffect.commandLine.SnpEffCmdEff.run(SnpEffCmdEff.java:939)
        at ca.mcgill.mcb.pcingola.snpEffect.commandLine.SnpEff.run(SnpEff.java:978)
        at ca.mcgill.mcb.pcingola.snpEffect.commandLine.SnpEff.main(SnpEff.java:136)
```
Thanks much for any help or suggestions on fixing or working around.

**Comment by chapmanb** [comment]
Sorry, I also forgot to mention that we saw this with human genome build 38. I could not reproduce with GRCh37, and both example command lines are included in the tarball. Thanks again.

**Comment by pcingola** [comment]
Solved in version 4.3

**Comment by chapmanb** [comment]
Pablo -- thanks much for this. I'll definitely do testing with 4.3. Much appreciated.
